### PR TITLE
Project code owners

### PR DESCRIPTION
**Pull request by srprash**
Issue: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/67

Adding code owners for the separate components of the contrib repo. Right now, all the code defaults to the @open-telemetry/dotnet-approvers group and I intend to keep it as such until we get specific approvers for each project. 
For the AWS code, I'm adding myself and @lupengamzn to be the code owners. Please feel free to discuss if you feel otherwise.

One thing to note is that any code owner added in this file will also need to have write permissions for the repo to actually let them be the code owners in a true sense.

### CODE REVIEW

File: CODEOWNERS
Comment (reyang) [inline CODEOWNERS:7]: I think this should be ```@open-telemetry/dotnet-contrib-approvers```?

File: CODEOWNERS
Comment (reyang) [inline CODEOWNERS:10]: Please sort alphabetically by default.

File: CODEOWNERS
Comment (srprash) [reply CODEOWNERS:7]: Since I'm not a member of the open-telemetry, I couldn't verify if the the `@open-telemetry/dotnet-contrib-approvers` has been created yet.
If it already exists or is going to be created soon, I can update these.

File: CODEOWNERS
Comment (cijothomas) [inline CODEOWNERS:14]: will have to validate if this is going to work, without both you being already have some permissions to this repo.

Comment (srprash) [reply CODEOWNERS:14]: I'm a bit confused by your comment. Do you want to validate if the code owners work without individuals being given write permission or do you want me to remove my self and Bolu from this file for now?

File: CODEOWNERS
Comment (SergeyKanzhelev) [inline CODEOWNERS:7]: is it necessary to duplicate `dotnet-contrib-approvers`? Just curious

**Review by SergeyKanzhelev — APPROVED** [review]

File: CODEOWNERS
Comment (srprash) [reply CODEOWNERS:7]: Actually I'm not 100% sure. I guess this will notify any person and also the `dotnet-contrib-approvers` team when the PR is created for relevant path files. In the absence of `dotnet-contrib-approvers`, the PR will only notify the persons and only default to `dotnet-contrib-approvers` if certain paths have unassigned people.

I referred to how the opentelemetry-collector-contrib manages its [codeowners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/.github/CODEOWNERS). If it works without the duplicate team mentions then we can remove it.

File: CODEOWNERS
Comment (cijothomas) [reply CODEOWNERS:14]: > Do you want to validate if the code owners work without individuals being given write permission 

this!

**Review by cijothomas — APPROVED** [review]

File: CODEOWNERS
Comment (srprash) [reply CODEOWNERS:14]: Okay. Then we can merge in this PR and I plan to create one today for adding CHANGELOG to AWS Project so we can test it then. 
However, the github [doc](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) does mention that:
> The people you choose as code owners must have write permissions for the repository. When the code owner is a team, that team must have write permissions, even if all the individual members of the team already have write permissions directly, through organization membership, or through another team membership.

File: CODEOWNERS
Comment (CodeBlanch) [inline CODEOWNERS:5]: This top rule applies to anything not defined? Should we then just define AWS stuff below and let everything else fall to the default?

Comment (cijothomas) [reply CODEOWNERS:5]: lets come back to this. will try merging this pr now to unblock progress.

**Pull request merged**
